src/main/java/org/cactoos/io/TailInput.java

```diff
@@ -0,0 +1,139 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import org.cactoos.Input;
+import org.cactoos.text.FormattedText;
+
+/**
+ * Input showing only last N bytes of the stream.
+ *
+ * <p>There is no thread-safety guarantee.
+ * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @version $Id$
+ * @since 0.30
+ */
+public final class TailInput implements Input {
+
+    /**
+     * Input to decorate.
+     */
+    private final Input input;
+
+    /**
+     * Number of last bytes to show from the input.
+     */
+    private final int count;
+
+    /**
+     * Maximum number of bytes to read at once.
+     */
+    private final int max;
+
+    /**
+     * Constructor.
+     * @param inpt Input to decorate
+     * @param bytes Number of last bytes to show from input
+     */
+    public TailInput(final Input inpt, final int bytes) {
+        // @checkstyle MagicNumber (1 line)
+        this(inpt, bytes, 16384);
+    }
+
+    /**
+     * Constructor.
+     * @param inpt Input to decorate
+     * @param bytes Number of last bytes to show from input
+     * @param maximum Maximum number of bytes to read at once
+     */
+    public TailInput(final Input inpt, final int bytes, final int maximum) {
+        this.input = inpt;
+        this.count = bytes;
+        this.max = maximum;
+    }
+
+    @Override
+    public InputStream stream() throws IOException {
+        if (this.max < this.count) {
+            throw new IllegalArgumentException(
+                new FormattedText(
+                    "Can't tail %d bytes if buffer is set to %d",
+                    this.count, this.max
+                ).asString()
+            );
+        }
+        final byte[] buffer = new byte[this.max];
+        final byte[] response = new byte[this.count];
+        int num = 0;
+        final InputStream stream = this.input.stream();
+        while (true) {
+            final int read = stream.read(buffer);
+            if (read <= 0) {
+                break;
```

Comment:
@krzyk I think you can remove 2 of 3 breaks here by using `else` part of `if`) It's hard to read code with a lot of return points.